packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -194,19 +205,22 @@ export default class GoogleCalendarService implements Calendar {
 
   async updateEvent(uid: string, event: CalendarEvent, externalCalendarId: string): Promise<any> {
     return new Promise(async (resolve, reject) => {
-      const [mainHostDestinationCalendar] =
-        event?.destinationCalendar && event?.destinationCalendar.length > 0 ? event.destinationCalendar : [];
+      const mainHostDestinationCalendar = event.destinationCalendar?.find(
+        (cal) => cal.externalId === externalCalendarId
+      );
```

Comment:
When rescheduling events we know the external id of the calendar so we can just look for it in the destinationCalendar array.

Reply:
This should be a comment in code